using_the_compiler/README.md

```diff
@@ -1,136 +1,322 @@
+
+* `--o <output_file>`: Define the name of the binary executable.
+* `--release`: Compile in release mode, doing extra work to apply optimizations to the generated code.
+* `--lto=thin`: Use ThinLTO, improving performance on release builds.
+* `--no-debug`: Skip any symbolic debug info, reducing the output file size.
+* `--progress`: Show progress during compilation.
+* `--static`: Link statically.
+
+More options are described in the integrated help: `crystal build --help` or man page `man crystal`.
+
+### `crystal eval`
+
+The `crystal eval` command reads Crystal source code from command line or stdin, compiles it to a binary executable and immediately runs it.
+
+```
+crystal eval [<options>] [<source>]
+```
+
+If no `source` argument is provided, the Crystal source is read from standard input. The binary executable is stored in a temporary location between compiling and running.
+
+Example:
+
+```shell
+$ crystal eval 'puts "Hello World"'
+Hello World!
+$ echo 'puts "Hello World"' | crystal eval
+Hello World!
 ```
 
-Make sure to always use `--release` for production-ready executables and when performing benchmarks.
+NOTE: When running interactively, stdin can usually be closed by typing the end of transmission character (`Ctrl+D`).
 
-The reason for this is that performance without full optimizations is still pretty good and provides fast compile times, so you can use the `crystal` command almost as if it were an interpreter.
+**Common Options:**
 
-To reduce the binary size, you can add the `--no-debug` flag and use the `strip` command. Debug symbols will be removed, use this option if only size is an issue and you won't need to debug the program.
+* `-o <output_file>`: Define the name of the binary executable.
+* `--release`: Compile in release mode, doing extra work to apply optimizations to the generated code.
+* `--lto=thin`: Use ThinLTO, improves performance.
+* `--no-debug`: Skip any symbolic debug info, reducing the output file size.
+* `--progress`: Show progress during compilation.
+* `--static`: Link statically.
 
-## Creating a statically-linked executable
+More options are described in the integrated help: `crystal build --help` or man page `man crystal`.
 
-To build a statically-linked executable of your program:
+### `crystal version`
+
+The `crystal version` command prints the Crystal version, LLVM version and default target triple.
 
 ```
-$ crystal build some_program.cr --release --static
+crystal version
 ```
 
-**Note:** Building statically-linked executables is currently only supported on Alpine Linux.
+Example:
 
-More information about statically linking [can be found on the wiki](https://github.com/crystal-lang/crystal/wiki/Static-Linking).
+```shell
+$ crystal version
+Crystal 0.25.1 [b782738ff] (2018-06-27)
 
-## Creating a project or library
+LLVM: 4.0.0
+Default target: x86_64-unknown-linux-gnu
+```
 
-Use the `init` command to create a Crystal project with the standard directory structure.
+### `crystal init`
```

Comment:
For someone who doesn't know what `crystal init` is, it's much more meaningful the title "Creating a project or library" than `crystal init`. How about combining both into: "Creating a project or library: `crystal init`"

Reply:
I'll add a new section at the top and leave the section about compiler commands named after the command.